### ISSUE TITLE
Version 1.1.0 started throwing java.lang.AbstractMethodError on startup

### ISSUE THREAD

**Issue by chansen-p44** [DataDog/dd-trace-java#4309]
Downloading version 1.1.0 from https://dtdg.co/latest-java-tracer results in the following error in a spring boot 2.5.12 application.
The previous version 1.0.1 works just fine.

```
24-Nov-2022 12:40:49.254 SEVERE [localhost-startStop-1] org.apache.catalina.core.StandardContext.filterStart Exception starting filter [servletPathRequestFilter]
	java.lang.AbstractMethodError: Receiver class datadog.trace.instrumentation.springweb.OrderedServletPathRequestFilter does not define or inherit an implementation of the resolved method 'abstract void init(javax.servlet.FilterConfig)' of interface javax.servlet.Filter.
		at org.apache.catalina.core.ApplicationFilterConfig.initFilter(ApplicationFilterConfig.java:281)
		at org.apache.catalina.core.ApplicationFilterConfig.<init>(ApplicationFilterConfig.java:109)
		at org.apache.catalina.core.StandardContext.filterStart(StandardContext.java:4607)
		at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5258)
		at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
		at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:753)
		at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:727)
		at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:695)
		at org.apache.catalina.startup.HostConfig.deployDirectory(HostConfig.java:1177)
		at org.apache.catalina.startup.HostConfig$DeployDirectory.run(HostConfig.java:1925)
		at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
		at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
		at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
		at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
		at java.base/java.lang.Thread.run(Thread.java:829)
24-Nov-2022 12:40:49.278 INFO [localhost-startStop-1] org.apache.catalina.core.ApplicationContext.log Closing Spring root WebApplicationContext
```

**Comment by amarziali** [comment]
Hi @chansen-p44 ,

we're trying to reproduce this issue but we had no success right now. It would be helpful to have more insight about the dependencies you are using. Would you mind sharing the dependency tree with us?

With maven: `mvn dependency:tree`

or with gradle: `gradle  dependencies` 

Thanks

Andrea

**Comment by amarziali** [comment]
We actually reproduced with an application server (like tomcat 8.5) running a servlet specification < 4. It should be fixed by #4312
